src/android/toga_android/fonts.py

```diff
@@ -37,22 +39,46 @@ def get_style(self):
         return Typeface.NORMAL
 
     def get_typeface(self):
-        if self.interface.family is SYSTEM:
-            family = Typeface.DEFAULT
-        elif self.interface.family is SERIF:
-            family = Typeface.SERIF
-        elif self.interface.family is SANS_SERIF:
-            family = Typeface.SANS_SERIF
-        elif self.interface.family is MONOSPACE:
-            family = Typeface.MONOSPACE
-        elif self.interface.family is CURSIVE:
-            family = Typeface.create("cursive", Typeface.NORMAL)
-        elif self.interface.family is FANTASY:
-            # Android appears to not have a fantasy font available by default,
-            # but if it ever does, we'll start using it. Android seems to choose
-            # a serif font when asked for a fantasy font.
-            family = Typeface.create("fantasy", Typeface.NORMAL)
-        else:
-            family = Typeface.create(self.interface.family, Typeface.NORMAL)
+        try:
+            family = _FONT_CACHE[self.interface]
+        except KeyError:
+            family = None
+            registered_font = self.interface.find_registered_font(
+                self.interface.family,
+                weight=self.interface.weight,
+                style=self.interface.style,
+                variant=self.interface.variant,
+            )
+            if registered_font is not None:
+                try:
+                    family = Typeface.createFromFile(
+                        str(self.interface.factory.paths.app / registered_font.path)
+                    )
+                except Exception as ex:
```

Comment:
This is a red flag; blanket exception handling can accidentally hide a wide range of problems. Can we narrow this down to a specific error?

Reply:
I can think of these cases: 
1. The font file does not exist
2. The file is a folder, not a file
3. The file is not a supported font file

What Python exceptions correspond to these cases?

How to find out if there are other cases?

In PrivateFontCollection, it says `If you try to use a font that is not supported, such as an unsupported OpenType font or a Bitmap font, an exception will occur. ` But what exception exactly?

Reply:
OK, I addressed this issue cleanly on Winforms.
On Android I can only handle the (crashing) file-not-found exception. The exception when using an invalid font file cannot be catched, but luckily, it does not crash the app

Reply:
I now catch the file-not-found exception when loading the font. Would it be better to catch this exception already in Font.register()? But then, register could not be a static method anymore because we need access to factory.paths.app